### ISSUE TITLE
Combination of merge and override behavior for configs

### ISSUE THREAD

**Issue by jemerald** [jenkinsci/templating-engine-plugin#97]
### Feature Request

We wanted to be able to provide default library configs in the governance tier, but allow the application layer to override them.

for example, with this setup:

```
// governance tier
someblock {
  config1 = "g1"
  config2 = "g2"
}

// application tier
someblock {
  config2 = "a2"
  config3 = "a3"
}
```

we would like to see the following config being applied:

```
someblock {
  config1 = "g1"
  config2 = "a2"
  config3 = "a3"
}
```

However this is not currently possible. With `merge = true`, config2 doesn't get overriden, and with `override = true`, config1 got lost.

**Comment by steven-terrana** [comment]
This feature will be available in JTE 2.0. 

I'll leave this open until that gets released, but the concept is that we're moving towards specifying merge and overriding using annotations so it can be done at the field level instead of just the block level. 

2.0:
```
// governance tier
@merge someblock {
  config1 = "g1"
  @override config2 = "g2"
}

// application tier
someblock {
  config2 = "a2"
  config3 = "a3"
}
```

results in: 
```
// aggregated config
someblock {
  config1 = "g1" 
  config2 = "a2" 
  config3 = "a3" 
}
```

I'll leave this open until 2.0 is released. 

Thanks for your feedback!

**Comment by steven-terrana** [comment]
Doing some spring cleaning, this functionality has been implemented and will be part of the 2.0 release.
